src/main/java/org/socialhistoryservices/delivery/request/controller/RequestController.java

```diff
@@ -99,27 +109,37 @@ public String scanBarcode(@RequestParam(required = false) String id, Model model
             return "request_scan";
         }
 
+        // list of all parents children excluding current holding
+        List<Holding> containerDependencies = records.getListOfSiblingHoldingInSameContainer(h);
+
         // Information about the current state
         Holding.Status oldStatus = h.getStatus();
         Request requestActive = requests.getActiveFor(h);
 
         // Determine the active request
         Reservation reservation = null;
         Reproduction reproduction = null;
-        if (requestActive instanceof Reservation)
+        if (requestActive instanceof Reservation) {
             reservation = (Reservation) requestActive;
-        if (requestActive instanceof Reproduction)
+        }
+        if (requestActive instanceof Reproduction) {
             reproduction = (Reproduction) requestActive;
+        }
 
         // Show the request corresponding to the scanned record
         if ((reservation != null) || (reproduction != null)) {
             // If the user may modify reservations, mark item for the active reservation
-            if (reservation != null)
+            if (reservation != null) {
                 reservations.markItem(reservation, h);
+                for (Holding containerDependancy : containerDependencies) {
```

Comment:
Why not do this on a lower level, so you don't repeat yourself later and prevents against bugs later when someone forgets to do this?